astroquery/mast/discovery_portal.py

```diff
@@ -116,23 +116,23 @@ class PortalAPI(BaseQuery):
     Should be used to facilitate all Portal API queries.
     """
 
+    MAST_REQUEST_URL = conf.server + "/api/v0/invoke"
+    COLUMNS_CONFIG_URL = conf.server + "/portal/Mashup/Mashup.asmx/columnsconfig"
+    MAST_DOWNLOAD_URL = conf.server + "/api/v0.1/Download/file"
+    MAST_BUNDLE_URL = conf.server + "/api/v0.1/Download/bundle"
+
+    TIMEOUT = conf.timeout
+    PAGESIZE = conf.pagesize
```

Comment:
Is there a particular reason of moving these out as class attributes as opposed as they were in ``__init__``? 


@keflavich - do we have one way or the other decided as a preference? I see that we mostly do `TIMEOUT` indeed as a class attribute, is that the way we decide to go forward?

Reply:
no, no established preference.

Reply:
I think the instance attribute makes more sense, e.g I see the need of changing the timeout, or cache preference, row limit, etc for one instance to allow e.g. a complicated query to run, but maybe not for another. I see that we had the TIMEOUT in the template as a class attribute, but we don't really have an init there, so I feel that should be fixed, too.

Having the URLs and other server specific config otoh sounds fine to be a class attribute, and indeed those are the original motivation of this PR, so to allow easier testing. (I think that can be done anyway, with the configuration system, but we already established that it's probably not the nicest API to have and thus are stepping away from it).

Reply:
Anyway, I don't think it should hold up this PR given many of the other modules have timeout as a class attribute. I'll suggest changing that separately.

Reply:
I initially had these under __init__ but realized my patchwork was getting overwritten when the class was re-initialized in methods under `astroquery.mast.Observations` for example.

Reply:
I'm not familiar with `astroquery.mast`. Can you point out some specific place in the code where such re-initialization happens?

Reply:
Here is one example: https://github.com/astropy/astroquery/blob/c933c3d6b927bca90cee61907c571a7021c9f070/astroquery/mast/observations.py#L121

where we use the PortalAPI class within `discovery_portal.py`

Reply:
As I said, I think it's OK to have the URLs and other server-specific items as class attributes to be able to change them once for all instances in a session. I don't see any user use case where they need to tinker with the URLs (but do correct me as I'm sure I don't think about all the scenarios), but having them as a class attribute would help testing. So, the majority of this PR is 👍 from me.

However, my issue is with doing the same with parameters that the user may want to change, and where there is a value in being able to change it on the instance level. E.g. timeout, row_limit, etc. But as I said above, this is an issue that affects much of the package, and therefore is something that we will want to clean up for an upcoming version that will have a lot of cleanup and probable API changes.

Reply:
Yep, I can bring this up in a team meeting. I've collapsed the commits down to 16, let me know if there is anything else that needs to be done.

Reply:
`astroquery` doesn't seem to handle configuration very well (#544 is as relevant now as it was when it was opened), but @bsipocz is correct in pointing out that this affects all of `astroquery` and is not specific to `mast`.

Reply:
OK, let's have this merged. The rest I put on a tentative agenda for a dev community meeting.